filters/openpolicyagent/internal/envoy/skipperadapter.go

```diff
@@ -11,6 +12,9 @@ import (
 )
 
 func AdaptToExtAuthRequest(req *http.Request, metadata *ext_authz_v3_core.Metadata, contextExtensions map[string]string, rawBody []byte) (*ext_authz_v3.CheckRequest, error) {
+	if err := validateURLForInvalidUTF8(req.URL); err != nil {
+		return nil, fmt.Errorf("invalid url: %w", err)
+	}
```

Comment:
Why do we need this check?

Reply:
This was done as a part of reducing errors in decision logs when the http path contained invalid UTF-8 characters.
(Introduced in https://github.com/zalando/skipper/pull/3086)

Reply:
Sorry should have been precise. I meant the decision logs in OPA.

Reply:
So do we want to save a roundtrip here by pre-validating URL?
Can this be done on OPA side, e.g. can it decide to reject request if URL is not valid?

Reply:
> So do we want to save a roundtrip here by pre-validating URL? Can this be done on OPA side, e.g. can it decide to reject request if URL is not valid?

To be honest, saving the round trip was just a side benefit. 

Our main concern was that requests with invalid URLs were failing before even reaching the policy logic (due to parse errors in OPA), but they still generated decision log entries marked as errors. This was creating a lot of noise in some of the application policies, especially when there were frequent malicious requests happening every day. We were worried that this noise could cause the teams to overlook or ignore real policy errors that actually need attention.